### PR TITLE
[chore] update builder to use non deprecated API

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -185,6 +185,7 @@ otelcorecol:
 .PHONY: genotelcorecol
 genotelcorecol:
 	pushd cmd/builder/ && $(GOCMD) run ./ --skip-compilation --config ../otelcorecol/builder-config.yaml --output-path ../otelcorecol && popd
+	$(MAKE) -C cmd/otelcorecol fmt
```
Comment (codeboten) [inline Makefile:188]: this seems unrelated to the change

Comment (bogdandrutu) [reply Makefile:188]: It is :)... the builder does not generate the imports in order, or at least I don't know how to do that in the template. because of that lint fails, so I decided that this is a nice fix.

Comment (bogdandrutu) [reply Makefile:188]: I can try to have:

```tmpl
	"go.opentelemetry.io/collector/component"
	"go.opentelemetry.io/collector/exporter"
	{{- range .Exporters}}
	{{.Name}} "{{.Import}}"
	{{- end}}
	"go.opentelemetry.io/collector/extension"
	{{- range .Extensions}}
	{{.Name}} "{{.Import}}"
	{{- end}}
```

This will work for core since exporters are all part of this repo so will start with go.opentelemetry.io/collector/exporter/... but does not work for contrib for example.

**Review by codeboten — APPROVED** [review]
